vllm/model_executor/model_loader/loader.py

```diff
@@ -541,7 +541,26 @@ def load_model(self, *, model_config: ModelConfig,
                                 "loading tensor of shape %s into "
                                 "parameter '%s' of shape %s", tensor.shape,
                                 key, param_shape)
-                        param_data.copy_(tensor)
+
+                        def maybe_set_scale(key, child=model):
+                            key_split = key.split('.', 1)
+                            if len(key_split) == 1:
+                                param = getattr(child, key_split[0])
+                                if hasattr(param, 'fp_quantizer') and param.use_meta_tensor:
+                                    fp8_data, scale = tensor.split(param.fp_quantizer.group_size, -1)
+                                    setattr(
+                                        param, 
+                                        'data', 
+                                        fp8_data.to(param.data.device).reshape(param.orig_shape[-1], param.orig_shape[-2])
+                                    )
+                                    setattr(param.fp_quantizer, 'scale', scale.to(param.data.device))
+                            else:
+                                maybe_set_scale(key_split[-1], getattr(child, key_split[0]))
+
+                        if param_data.dtype == torch.uint8:
+                            maybe_set_scale(key)
+                        else:
+                            param_data.copy_(tensor)
```

Comment:
Could you handle this within the quantization file through implementing a `DeepSpeedFPLinearMethod.process_weights_after_loading`? 

Example here for fp8: https://github.com/vllm-project/vllm/blob/461089a21a5b00d6c6712e3bf371ce2d9cfa0860/vllm/model_executor/layers/quantization/fp8.py#L162